iothub/device/src/IDelegatingHandler.cs

```diff
@@ -15,6 +15,7 @@ interface IDelegatingHandler : IContinuationProvider<IDelegatingHandler>, IDispo
         Task OpenAsync(CancellationToken cancellationToken);
         Task CloseAsync(CancellationToken cancellationToken);
         Task WaitForTransportClosedAsync();
+        bool IsReady { get; }
 
         // Telemetry uplink.
```

Comment:
Quick question: why isn't it possible to close before handing over the control to these layers from mqtt layer?

Reply:
Yes, we can close, but the upper layers won't be aware of it..